.github/buildomat/jobs/check-features.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+#:
+#: name = "check-features (helios)"
+#: variety = "basic"
+#: target = "helios-2.0"
+#: rust_toolchain = true
+#: output_rules = []
+
+# Run cargo check on illumos with feature-specifics like `no-default-features`
+# or `feature-powerset`.
+
+set -o errexit
+set -o pipefail
+set -o xtrace
+
+cargo --version
+rustc --version
+
+# NOTE: This version should be in sync with the recommended version in
+# ./dev-tools/xtask/src/check-features.rs.
+CARGO_HACK_VERSION='0.6.28'
+
+#
+# Set up our PATH for use with this workspace.
+#
+source ./env.sh
+
+banner prerequisites
+ptime -m bash ./tools/install_builder_prerequisites.sh -y
+
+banner check
+export CARGO_INCREMENTAL=0
+ptime -m cargo check --workspace --bins --tests --no-default-features
+RUSTDOCFLAGS="--document-private-items -D warnings" ptime -m cargo doc --workspace --no-deps --no-default-features
+
+#
+# Check the feature set with the `cargo xtask check-features` command.
+#
+banner hack
+ptime -m timeout 2h cargo xtask check-features --version "$CARGO_HACK_VERSION" --exclude-features image-trampoline,image-standard
```

Comment:
So what I'd like is for all of this `--exclude-features` configuration to be part of the xtask as well.

Basically, think of how a developer who runs into this CI failure would address this. They're not going to be running `cargo xtask check-features` as part of their normal workflows, so it's going to be a little surprising to them. But then they'll have to learn to open up this file and copy over this command line.

Instead, if you provided a simple `cargo xtask check-features --ci --install-version 0.6.28` which does all this setup for you, then that's a simple command they can run.

(This is what inspired nextest's [support for configuration profiles](https://nexte.st/docs/configuration/#profiles) -- CI should contain as little logic as possible so that non-CI users can replicate most or all of it locally.)

Reply:
Yeah, I did this first, but the exclusions were pretty specific to those image-features, which I admittedly don't know much about. I was wondering if such compile-features were going to happen more often, but probably not.